Update for compatibility with Phoenix 1.3 and latest elixir.

tokafish opened the pull request:
First off, thanks for the work on this package!

I attempted to use this with a brand new Phoenix app. I found some issues which I believe I've corrected:

* I made `:router` configurable via the application env, since the new default in Phoenix 1.3 is AppName**Web**.Router, rather than `AppName.Router`.
* I fixed a deprecation on `Mix.Utils.camelize`
* Formatted and fixed warnings.

It works now on my app, and I believe it's still compatible with older versions.

tiagoengel commented:
Thanks for the PR @tokafish! I've been meaning to do this for a while!

A few things:

- This doesn't need to be compatible with older versions of Elixir/Phoenix. I plan to release it as a breaking change so people have to opt in to update. If they want to use it with older versions they can use the current version. Having said that, I think the router change is not necessary and we should update it to reflect the new default in Phoenix.
- Tests are failing now, seems like a problem with the lib used to parse and execute javascript code during tests.
- We need to update the example app to use the latest Phoenix version - This app is also used for integration tests so we need to make sure it's passing with the new version.

tokafish commented:
I updated the PR to use Phoenix's new default module naming, and updated the example app and tests. Live reload appears to work when I boot phoenix within `example` and edit the router file, although the the example app's test doesn't seem to agree. Not sure if something is misconfigured in the test env, or if it's hound's fault. If you have time to take a brief look, I'd appreciate it!

tiagoengel commented:
Seems like a problem with `hackney` calling `erlang.send_after(:infinity, #PID<0.436.0>, {:timeout, #Port<0.10277>})` with `:infinity`. see: http://erlang.org/doc/man/erlang.html#send_after-3 and http://erlang.org/doc/man/erlang.html#send_after-3

Not sure how to fix it, haven't got a lot of time to play with it to be honest.

There's also a problem with live reload and ecto happening in the CI:

<img width="939" alt="screen shot 2018-07-23 at 21 34 52" src="https://user-images.githubusercontent.com/1457263/43101456-57f4b290-8ec0-11e8-93c1-da4cc48e4e84.png">

I'm wondering if it wouldn't be easier to just delete the current example app and create a fresh one with the phoenix cli.

tokafish commented:
I did spin up a new example. I switched the dependency on hound to wallaby, which solved the hackney problem with `send_after`. The live reload test still fails, although I can see it work properly in the real server. When I debug what the compiler's doing, I see that the module hash isn't changing as expected after the test changes the file.

tiagoengel commented on example/lib/js_routes_example_web/router.ex:
```diff
@@ -0,0 +1,39 @@
+  end
+
+  scope "/test", JsRoutesExampleWeb do
+    # TEST-PLACEHOLDER#
```
tests are failing because of this, it should be `#TEST-PLACEHOLDER#`

tiagoengel commented on example/lib/js_routes_example_web/router.ex:
```diff
@@ -0,0 +1,39 @@
+  end
+
+  scope "/test", JsRoutesExampleWeb do
+    # TEST-PLACEHOLDER#
```
tests are failing because of this. it should be `#TEST-PLACEHOLDER#`

tokafish replied:
Thanks, that was my problem (new elixir formatter!)

tiagoengel commented on example/test/js_routes_example_web/integration/phoenix_jsroutes_test.exs:
```diff
@@ -0,0 +1,76 @@
+
+    new_routes = "
+    get \"user/:id\", TestUserController, :index
+    post \"user/:userId/projects/:id\", TestUserProjectsController, :update
```
both lines here need a forward slash at the beginning. i.e. `get  \"/user/:id\", TestUserController, :index`

tiagoengel commented on example/priv/static/js/app.js:
```diff
@@ -1,25 +1,86 @@
-var container = document.getElementById('routes-container');
+/******/ (function(modules) { // webpackBootstrap
```
This is not right. This is the "compiled" version, we want the original version here.

tokafish replied:
original should be in `/example/assets` right?

tiagoengel replied:
oh that's right! We need to ignore those two folders (`example/priv/static/css` and `example/priv/static/js`) and remove the files from your commit, as they are compiled files.

tiagoengel commented on example/assets/js/socket.js:
```diff
@@ -0,0 +1,63 @@
+// NOTE: The contents of this file will only be executed if
```
Can we remove this?

tokafish commented:
I think this is ready to go, but let me know what else you'd like to see changed!
